nc/source/component/AutoComponentGroup.cpp

```diff
@@ -0,0 +1,54 @@
+#include "component/AutoComponentGroup.h"
+
+namespace nc
+{
+    AutoComponentGroup::AutoComponentGroup(Entity)
+        : m_components{}
+    {}
+
+    auto AutoComponentGroup::GetAutoComponents() const noexcept -> std::vector<AutoComponent*>
+    {
+        std::vector<AutoComponent*> out(m_components.size());
+        std::transform(m_components.cbegin(), m_components.cend(), out.begin(), [](const auto& ptr)
+        {
+            return ptr.get();
+        });
+        return out;
+    }
+
+    void AutoComponentGroup::SendFrameUpdate(float dt)
+    {
+        for(auto& comp : m_components)
+            comp->FrameUpdate(dt);
+    }
+
+    void AutoComponentGroup::SendFixedUpdate()
+    {
+        for(auto& comp : m_components)
+            comp->FixedUpdate();
+    }
+
+    void AutoComponentGroup::SendOnDestroy()
+    {
+        for(auto& comp : m_components)
+            comp->OnDestroy();
+    }
+
+    void AutoComponentGroup::SendOnCollisionEnter(Entity hit)
```

Comment:
I don't like the collision specific methods here. Will these be moved in future PRs?